ui/src/ducks/app/solutions.js

```diff
@@ -143,45 +143,64 @@ export function* prepareEnvironment(action) {
     console.error(`Environment '${envName}' is preparing`);
   }
 
-  const addSolutionToEnvironmentResult = yield call(
-    SolutionsApi.addSolutionToEnvironment,
-    envName,
-    solName,
-    solVersion,
+  // Need to check if Salt is under `deploy-node` before adding solution to environment
```

Comment:
I don't think that's the issue with #2941 - "solution import" means running `./solutions.sh import --archive /path/to/solution.iso` from the Bootstrap node.
Adding a Solution to an environment only means performing K8s API calls (either from the UI directly, or from Salt master), but it should **not** trigger any Salt master restart.

So I don't think this check is needed at all.

Reply:
Absolutely right, I was confused with solution import and add a solution to an environment.
I will remove this checking...